test/modules/EdgeHubRestartTester/src/Program.cs

```diff
@@ -37,16 +38,32 @@ static async Task<int> MainAsync()
 
                 if (Settings.Current.MessageEnabled)
                 {
+                    ModuleClient msgModuleClient = await ModuleUtil.CreateModuleClientAsync(
+                        Settings.Current.TransportType,
+                        ModuleUtil.DefaultTimeoutErrorDetectionStrategy,
+                        ModuleUtil.DefaultTransientRetryStrategy,
+                        Logger);
+
+                    msgModuleClient.OperationTimeoutInMilliseconds = (uint)Settings.Current.SdkOperationTimeout.TotalMilliseconds;
+
                     edgeHubMessageConnector = new MessageEdgeHubConnectorTest(
                         batchId,
-                        Logger);
+                        Logger,
+                        msgModuleClient);
                 }
 
                 if (Settings.Current.DirectMethodEnabled)
                 {
+                    ModuleClient dmModuleClient = await ModuleUtil.CreateModuleClientAsync(
+                        Settings.Current.TransportType,
+                        ModuleUtil.DefaultTimeoutErrorDetectionStrategy,
+                        ModuleUtil.DefaultTransientRetryStrategy,
+                        Logger);
+
```

Comment:
I think you need to set OperationTimeoutInMilliseconds.  Please extract module client creation code as a method and reuse to create module client for messaging and DM scenario.

Reply:
DM is SDK timeout is set upon request. 
Message SDK timeout is set upon client creation.

Reply:
Thanks for explanation. @davilu would you please confirm that OperationTimeoutInMilliseconds is not used for Direct Method call?